File: replace str in directory files/php/replace.php
<?php

function replaceInFile($filePath, $search, $replace) {
    $fileContent = file_get_contents($filePath);
    

    if (strpos($fileContent, $search) !== false) {
        $newContent = str_replace($search, $replace, $fileContent);
        
        if ($newContent !== $fileContent) {
            file_put_contents($filePath, $newContent);
            return true; 
        }
    }
    return false;
}

function recursiveSearch($directory, $search, $replace) {
    $modifiedFiles = []; 
    $dir = opendir($directory);
    
    while (($file = readdir($dir)) !== false) {
        if ($file != '.' && $file != '..') {
            $filePath = $directory . '/' . $file;
            if (is_dir($filePath)) {
                $modifiedFiles = array_merge($modifiedFiles, recursiveSearch($filePath, $search, $replace));
            } else {
                if (replaceInFile($filePath, $search, $replace)) {
                    $modifiedFiles[] = $filePath;
                }
            }
        }
    }
    closedir($dir);
    return $modifiedFiles;
}


$directoryPath = readline("Enter the directory path: ");
$searchWord = readline("Enter the word to search: ");
$replaceWord = readline("Enter the word to replace with: ");


if (!is_dir($directoryPath)) {
    echo "The specified directory does not exist.\n";
    exit(1);
}

$modifiedFiles = recursiveSearch($directoryPath, $searchWord, $replaceWord);

if (!empty($modifiedFiles)) {
    echo "Replacements were made in the following files:\n";
    foreach ($modifiedFiles as $modifiedFile) {
        echo $modifiedFile . "\n";
    }
} else {
    echo "No replacements were made.\n";
}
?>

File: replace str in directory files/python/replace.py
import os

def replace_in_file(file_path, search, replace):
    with open(file_path, 'r', encoding='utf-8') as file:
        file_content = file.read()

    if search in file_content:
        new_content = file_content.replace(search, replace)
        
        if new_content != file_content:
            with open(file_path, 'w', encoding='utf-8') as file:
                file.write(new_content)
            return True
    return False

def recursive_search(directory, search, replace):
    modified_files = []
    
    for root, dirs, files in os.walk(directory):
        for file in files:
            file_path = os.path.join(root, file)
            if replace_in_file(file_path, search, replace):
                modified_files.append(file_path)
    
    return modified_files


directory_path = input("Enter the directory path: ")
search_word = input("Enter the word to search: ")
replace_word = input("Enter the word to replace with: ")


if not os.path.isdir(directory_path):
    print("The specified directory does not exist.")
    exit(1)

modified_files = recursive_search(directory_path, search_word, replace_word)

if modified_files:
    print("Replacements were made in the following files:")
    for modified_file in modified_files:
        print(modified_file)
else:
    print("No replacements were made.")
